js/components/controlBar.js

```diff
@@ -146,77 +146,80 @@ var ControlBar = React.createClass({
   },
 
   handleQualityClick: function () {
-    if (this.props.controller.state.videoQualityOptions.showVideoQualityPopover) {
-      // Reset autoFocus property when closing the quality menu
-      this.props.controller.state.videoQualityOptions.autoFocus = false;
-    } else {
-      // If the quality menu was activated via keyboard we should
-      // autofocus on the first element
-      this.props.controller.state.videoQualityOptions.autoFocus = this.qualityMenuOpenedWithKeyboard;
-    }
+    this.configureMenuAutofocus(CONSTANTS.MENU_OPTIONS.VIDEO_QUALITY);
 
     if (this.props.responsiveView == this.props.skinConfig.responsive.breakpoints.xs.id) {
       this.props.controller.toggleScreen(CONSTANTS.SCREEN.VIDEO_QUALITY_SCREEN);
     } else {
-      this.toggleQualityPopover();
+      this.togglePopover(CONSTANTS.MENU_OPTIONS.VIDEO_QUALITY);
       this.closeCaptionPopover();
     }
   },
 
-  /**
-   * Fires when a key is pressed on the video quality button.
-   * @private
-   * @param {type} event The keydown event object.
-   */
-  handleQualityKeyDown: function(event) {
-    switch (event.key) {
-      case CONSTANTS.KEY_VALUES.SPACE:
-      case CONSTANTS.KEY_VALUES.ENTER:
-      // Ctrl and Alt are needed as a workaround for VoiceOver, which uses the
-      // CTRL + OPTION + SPACE combination to activate buttons. VoiceOver actually
-      // suppresses the spacebar keyboard event when this combination is used, so we
-      // can only detect either CTRL or OPTION. This can obviously fail if the user
-      // presses a different key after CTRL + OPTION, but a false positive is preferred.
-      case CONSTANTS.KEY_VALUES.CONTROL:
-      case CONSTANTS.KEY_VALUES.ALT:
-        this.qualityMenuOpenedWithKeyboard = true;
-        break;
-      default:
-        break;
+  handleClosedCaptionClick: function () {
+    this.configureMenuAutofocus(CONSTANTS.MENU_OPTIONS.CLOSED_CAPTIONS);
+
+    if (this.props.responsiveView == this.props.skinConfig.responsive.breakpoints.xs.id) {
+      this.props.controller.toggleScreen(CONSTANTS.SCREEN.CLOSEDCAPTION_SCREEN);
+    } else {
+      this.togglePopover(CONSTANTS.MENU_OPTIONS.CLOSED_CAPTIONS);
+      this.closeQualityPopover();
     }
   },
 
-  toggleQualityPopover: function () {
-    if (this.props.controller.state.videoQualityOptions.showVideoQualityPopover) {
-      this.qualityMenuOpenedWithKeyboard = false;
+  configureMenuAutofocus: function(menu) {
+    var menuOptions = this.props.controller.state[menu] || {};
```

Comment:
Should we just return if this.props.controller.state[menu] is falsy? I don't see anything below impact anything outside the scope of this function.

Reply:
Yes I actually changed it in order to save a few lines of code. I can add the return if you prefer it.

Reply:
Sounds good, we can leave it as is.